service/service.go

```diff
@@ -213,6 +165,35 @@ func (t *service) Store() lstore.Logstore {
 	return t.store
 }
 
+// GetHostID returns the host's peer ID.
+func (t *service) GetHostID(context.Context) (peer.ID, error) {
+	return t.host.ID(), nil
+}
+
+// CreateThread with id.
+func (t *service) CreateThread(_ context.Context, id thread.ID) (info thread.Info, err error) {
+	info.ID = id
+	info.FollowKey, err = sym.CreateKey()
+	if err != nil {
+		return
+	}
+	info.ReadKey, err = sym.CreateKey()
+	if err != nil {
+		return
+	}
+	if err = t.store.AddThread(info); err != nil {
+		return
+	}
+	linfo, err := createLog(t.host.ID())
+	if err != nil {
+		return
+	}
+	if err = t.store.AddLog(id, linfo); err != nil {
```

Comment:
Now that creating a thread is actually part of the API, I added this to ensure that a log is created for the host at creation time. Not having this always felt a little weird, and I got bit by it again debugging a new AddFollower test. This would fail silently if the creator hadn't added a record yet, because there was not logs to send the follower.